packages/user-mgnt/src/features/countUsersByLocation/service.ts

```diff
@@ -26,16 +28,26 @@ export async function countUsersByLocation(
     ])
     return resArray[0] ?? { registrars: 0 }
   }
+
+  const locationChildren = await resolveLocationChildren(locationId)
+
   const resArray = await User.aggregate([
     {
       $match: {
-        catchmentAreaIds: locationId,
+        primaryOfficeId: { $in: locationChildren },
         systemRole
       }
     },
-    { $unwind: '$catchmentAreaIds' },
-    { $group: { _id: '$catchmentAreaIds', registrars: { $sum: 1 } } },
-    { $match: { _id: locationId } }
+    {
+      $group: { _id: '$primaryOfficeId', registrars: { $addToSet: '$_id' } }
+    },
+    {
+      $project: {
+        _id: 0,
+        registrars: { $size: '$registrars' } // Count the number of unique user ids collected in registrars
+      }
+    }
   ])
+
```

Comment:
This should be tested to add confidence to these calculations

Reply:
Just a bit difficult as we currently don't have a way of mocking the database itself, only mocking the results of these functions with mockingoose 🤔